spec/facility_spec.rb

```diff
@@ -23,4 +32,189 @@
       expect(@facility.services).to eq(['New Drivers License', 'Renew Drivers License', 'Vehicle Registration'])
     end
   end
-end
+
+  describe '#register_vehicle' do
+    before(:each) do
+      @facility.add_service('Vehicle Registration')
+    end
+
+    it 'can register an antique vehicle' do
+      @facility.add_service('Vehicle Registration')
+      expect(@facility.registered_vehicles).to eq([])
+      expect(@facility.collected_fees).to eq(0)
+      expect(@antique_car.registration_date).to eq(nil)
+      expect(@antique_car.plate_type).to eq(nil)
+      @facility.register_vehicle(@antique_car)
+      expect(@facility.registered_vehicles).to eq([@antique_car])
+      expect(@facility.collected_fees).to eq(25)
+      expect(@antique_car.registration_date).to eq(Date.today)
+      expect(@antique_car.plate_type).to eq(:antique)
```

Comment:
Lots of side effect testing in here! This could be also tested via a few smaller tests, but I like how this is working.